omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/visit/RetrospectiveVisitFragmentController.java

```diff
@@ -36,16 +36,9 @@ public Object create(@SpringBean("adtService") AdtService adtService,
             stopDate = startDate;
         }
 
-        // set the startDate time component to the start of day
-        startDate = new DateTime(startDate).withTime(0,0,0,0).toDate();
+        //If stopDate is today, it cannot be today at 23:59:59, because that would be in the future.
+        stopDate = stopDate.after(new Date()) ? new Date() : stopDate;
 
-        // if stopDate is today, set stopDate to current datetime, otherwise set time component to end of date
```

Comment:
Aren't you altering the original behavior even when ui.convertTimezones may not be set?

Reply:
We changed the way that we format the start/end date, but in the end, everything should be the same as before
Before we sent 2 dates from the client, and then we format on the server side to be startDate 00:00:00 and endDate 23:59:59:59. 
Now we are formatting the 00:00:00 and 23:59:59 on the client side. [here](https://github.com/openmrs/openmrs-module-coreapps/pull/394/files/3adf321846668e81a3048ac3711f8f97e7c72d27#diff-3b4bdfce9ba945b655e5733148593aef49f21e4e31c59f063115c92015d24555R59-R60), this way, if we are using timezone, we can send the date correctly formatted with timezone, if we are not using timezones, we will send the information without timezone, [here](https://github.com/openmrs/openmrs-module-coreapps/pull/394/files/3adf321846668e81a3048ac3711f8f97e7c72d27#diff-3b4bdfce9ba945b655e5733148593aef49f21e4e31c59f063115c92015d24555R66-R68)

Reply:
Are the date formatting changes related to time zones?

Reply:
Yes, when we use timezones, we send datetime and timezone, when we dont use tz we only send datetime

Reply:
How about doing the formatting only when the time zone global property is set? That way, we avoid the risk of introducing bugs for those who just want the old behaviour and do not care about time zones.

Reply:
I agree with @dkayiwa as I'd prefer not to touch this code if the property is not set, but if it would make things really ugly, I guess I could live with it... we just need to make sure we test fully.